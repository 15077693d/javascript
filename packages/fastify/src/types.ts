import type { ClerkOptions } from '@clerk/backend';

export type ClerkFastifyOptions = Omit<ClerkOptions, 'apiKey'>;
