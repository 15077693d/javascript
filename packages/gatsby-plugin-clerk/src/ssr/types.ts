import { Session, User } from '@clerk/backend-core/src';
import type { GetServerDataProps } from "gatsby"
import { ServerGetToken } from '@clerk/types';

export type GetServerDataPropsWithAuth = GetServerDataProps & {
  auth?: {
    session: Session | undefined | null;
    userId: string | null;
    sessionId: string | null;
    getToken: ServerGetToken
  };
  user?: User | undefined | null;
}
