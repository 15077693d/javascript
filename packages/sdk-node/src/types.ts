import type { AuthenticateRequestOptions, AuthObject, Clerk, SignedInAuthObject } from '@clerk/backend';
import type { MultiDomainAndOrProxy } from '@clerk/types';
import type { NextFunction, Request, Response } from 'express';
import type { IncomingMessage } from 'http';

type LegacyAuthObject<T extends AuthObject> = Pick<T, 'sessionId' | 'userId' | 'actor' | 'getToken' | 'debug'> & {
  claims: AuthObject['sessionClaims'];
};

type BooleanOrURLFnToBoolean = boolean | ((url: URL) => boolean);

export type MiddlewareWithAuthProp = (
  // req: WithAuthProp<Request>
  req: Request,
  res: Response,
  next: NextFunction,
) => Promise<void>;

export type MiddlewareRequireAuthProp = (
  // req: RequireAuthProp<Request>,
  req: Request,
  res: Response,
  next: NextFunction,
) => Promise<void>;

export type LooseAuthProp = { auth: LegacyAuthObject<AuthObject> };
export type StrictAuthProp = { auth: LegacyAuthObject<SignedInAuthObject> };
export type WithAuthProp<T> = T & LooseAuthProp;
export type RequireAuthProp<T> = T & StrictAuthProp;

export type ClerkMiddleware = MiddlewareWithAuthProp | MiddlewareRequireAuthProp;
export type ClerkMiddlewareOptions = {
  onError?: (error: any) => unknown;
  authorizedParties?: string[];
  audience?: AuthenticateRequestOptions['audience'];
  jwtKey?: string;
  strict?: boolean;
  signInUrl?: string;
  /**
   * @experimental
   */
  experimental_isApiRoute?: BooleanOrURLFnToBoolean;
} & MultiDomainAndOrProxy;

export type ClerkClient = ReturnType<typeof Clerk>;

export type AuthenticateRequestParams = {
  clerkClient: ClerkClient;
  publishableKey?: string;
  secretKey?: string;
  /**
   * @deprecated Use `publishableKey` instead.
   */
  frontendApi?: string;
  /**
   * @deprecated Use `secretKey` instead.
   */
  apiKey?: string;
  req: IncomingMessage;
  options?: ClerkMiddlewareOptions;
};
