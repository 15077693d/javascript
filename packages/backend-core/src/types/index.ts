export * from './jwt';
export * from './core';
