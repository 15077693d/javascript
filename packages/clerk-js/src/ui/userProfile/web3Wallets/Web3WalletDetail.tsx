// @ts-ignore
import { default as BinIcon } from '@clerk/shared/assets/icons/bin.svg';
// @ts-ignore
import { default as MailIcon } from '@clerk/shared/assets/icons/mail.svg';
import { Button } from '@clerk/shared/components/button';
import { List } from '@clerk/shared/components/list';
import { VerificationStatusTag } from '@clerk/shared/components/tag';
import { TitledCard } from '@clerk/shared/components/titledCard';
import { Toggle } from '@clerk/shared/components/toggle';
import React from 'react';
import { handleError } from 'ui/common';
import { Alert } from 'ui/common/alert';
import { useCoreUser } from 'ui/contexts';
import { useNavigate } from 'ui/hooks';
import { useRouter } from 'ui/router';
import { PageHeading } from 'ui/userProfile/pageHeading';

import { EditableListFieldRemoveCard } from '../EditableListFieldRemoveCard';
import { PrimaryTag } from '../util';

export function Web3WalletDetail(): JSX.Element | null {
  const { navigate } = useNavigate();
  const user = useCoreUser();
  const { params } = useRouter();
  const [showRemovalPage, setShowRemovalPage] = React.useState<boolean>(false);
  const [error, setError] = React.useState<string | undefined>();

  const web3Wallet = user.web3Wallets.find(ea => ea.id === params.web3_wallet_id);
  const isPrimary = web3Wallet?.id === user.primaryWeb3WalletId;

  if (!web3Wallet) {
    return null;
  }

  const verificationStatus = web3Wallet.verification?.status || 'unverified';
  const isVerified = web3Wallet.verification?.status === 'verified';

  const handleRemove = () => {
    setShowRemovalPage(true);
  };

  const onWeb3WalletRemove = () => {
    return web3Wallet.destroy();
  };

  const makeIdentPrimary = async () => {
    await user.update({ primaryWeb3WalletId: web3Wallet.id }).catch(e => handleError(e, [], setError));
  };

  const removeWeb3WalletScreen = (
    <EditableListFieldRemoveCard
      type='web3_wallet'
      label={web3Wallet.web3Wallet}
      buttonLabel='Unlink'
      onCancel={() => {
        setShowRemovalPage(false);
      }}
      onRemove={onWeb3WalletRemove}
      onRemoved={() => {
        navigate('../');
      }}
    />
  );

  return (
    <>
      <PageHeading
        title='Web3 wallet'
        subtitle={isVerified ? 'Manage this Web3 wallet' : 'Verify this Web3 wallet'}
        backTo='./../'
      />
      {showRemovalPage && removeWeb3WalletScreen}
      {!showRemovalPage && (
        <TitledCard className='cl-themed-card cl-list-card cl-verifiable-field-card'>
          <Alert
            type='error'
            style={{ margin: '0 2em' }}
          >
            {error}
          </Alert>

          <List>
            <List.Item
              className='cl-list-item'
              hoverable={false}
              detail={false}
              lines
            >
              <div>
                <div className='cl-ident-detail'>
                  {web3Wallet.web3Wallet}
                  <div className='cl-tags'>
                    {isPrimary && <PrimaryTag />}
                    <VerificationStatusTag
                      className='cl-tag'
                      status={verificationStatus}
                    />
                  </div>
                </div>
              </div>
            </List.Item>

            {isVerified && (
              <List.Item
                className='cl-list-item'
                hoverable={false}
                detail={false}
                lines
              >
                <div className='cl-primary-status-container'>
                  <div className='cl-description'>
                    <MailIcon />
                    <div className='cl-text'>
                      <div className='cl-title'>Primary web3 wallet</div>
                    </div>
                  </div>
                  <Toggle
                    name='primary_toggle'
                    disabled={isPrimary}
                    checked={isPrimary}
                    handleChange={makeIdentPrimary}
                  />
                </div>
              </List.Item>
            )}
          </List>
          <Button
            onClick={handleRemove}
            className='cl-add-resource-button cl-remove-button'
            type='button'
            flavor='text'
            buttonColor='error'
            hoverable
          >
            <BinIcon />
            Unlink
          </Button>
        </TitledCard>
      )}
    </>
  );
}
