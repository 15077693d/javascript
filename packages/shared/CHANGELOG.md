# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

### [0.0.16-alpha.4](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-alpha.3...@clerk/shared@0.0.16-alpha.4) (2022-03-23)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.3](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-alpha.2...@clerk/shared@0.0.16-alpha.3) (2022-03-22)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.2](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.2) (2022-03-22)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.1](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.1) (2022-03-20)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.0) (2022-03-19)

**Note:** Version bump only for package @clerk/shared

### [0.0.15](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.15-staging.1...@clerk/shared@0.0.15) (2022-03-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.15-staging.1](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.15-staging.0...@clerk/shared@0.0.15-staging.1) (2022-03-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.13](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.12...@clerk/shared@0.0.13) (2022-03-11)

**Note:** Version bump only for package @clerk/shared

### [0.0.12](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.12-staging.0...@clerk/shared@0.0.12) (2022-03-09)

**Note:** Version bump only for package @clerk/shared

### [0.0.10](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.9...@clerk/shared@0.0.10) (2022-03-04)

**Note:** Version bump only for package @clerk/shared

### [0.0.9](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.8...@clerk/shared@0.0.9) (2022-03-04)

**Note:** Version bump only for package @clerk/shared

### [0.0.8](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.7...@clerk/shared@0.0.8) (2022-03-03)

**Note:** Version bump only for package @clerk/shared

### [0.0.7](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.6...@clerk/shared@0.0.7) (2022-03-02)

**Note:** Version bump only for package @clerk/shared

### [0.0.6](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.5...@clerk/shared@0.0.6) (2022-03-01)

### Bug Fixes

- **shared:** Phone input should start blank ([43f0b06](https://github.com/clerkinc/clerk_docker/commit/43f0b0603608866f6b9e0a37a284c0ea72c0004b))

### [0.0.5](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.5-staging.0...@clerk/shared@0.0.5) (2022-02-24)

**Note:** Version bump only for package @clerk/shared

### [0.0.5-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.4-staging.0...@clerk/shared@0.0.5-staging.0) (2022-02-24)

**Note:** Version bump only for package @clerk/shared

### [0.0.4-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.3-staging.0...@clerk/shared@0.0.4-staging.0) (2022-02-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.3-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.2...@clerk/shared@0.0.3-staging.0) (2022-02-15)

**Note:** Version bump only for package @clerk/shared

### [0.0.2](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.2-staging.0...@clerk/shared@0.0.2) (2022-02-14)

**Note:** Version bump only for package @clerk/shared

### 0.0.2-staging.0 (2022-02-11)

**Note:** Version bump only for package @clerk/shared
