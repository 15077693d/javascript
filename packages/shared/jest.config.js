/** @type {import('ts-jest').JestConfigWithTsJest} */
const config = {
  displayName: 'shared',
  injectGlobals: true,

  testEnvironment: 'jsdom',
  roots: ['<rootDir>/src'],
  setupFiles: ['./setupJest.ts'],
  testRegex: ['/ui/.*/__tests__/.*.test.[jt]sx?$', '/(core|utils)/.*.test.[jt]sx?$'],
  testPathIgnorePatterns: ['/node_modules/'],

  collectCoverage: false,
  coverageProvider: 'v8',
  coverageDirectory: 'coverage',
  coveragePathIgnorePatterns: ['/node_modules/'],

  moduleDirectories: ['node_modules', '<rootDir>/src'],
  transform: {
    '^.+\\.m?tsx?$': ['ts-jest', { tsconfig: 'tsconfig.test.json', diagnostics: false }],
  },
};

module.exports = config;
