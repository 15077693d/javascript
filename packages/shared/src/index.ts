/** The following files are not exported on purpose:
 * - cookie.ts
 * - globs.ts
 *
 * The following folders are also not exported on purpose:
 * - error
 * - react
 * - utils
 *
 * People should always use @clerk/shared/<name> instead
 */

export { createWorkerTimers } from './workerTimers';
export * from './browser';
export * from './color';
export * from './date';
export * from './deprecated';
export * from './file';
export { isomorphicAtob } from './isomorphicAtob';
export * from './keys';
export { loadScript } from './loadScript';
export { LocalStorageBroadcastChannel } from './localStorageBroadcastChannel';
export * from './poller';
export * from './proxy';
export * from './underscore';
export * from './url';
