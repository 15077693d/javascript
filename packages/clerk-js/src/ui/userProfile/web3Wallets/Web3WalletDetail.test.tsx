import { renderJSON } from '@clerk/shared/testUtils';
import { EnvironmentResource, UserResource, UserSettingsResource, Web3WalletResource } from '@clerk/types';
import React from 'react';
import { PartialDeep } from 'type-fest';

import { Web3WalletDetail } from './Web3WalletDetail';

const mockNavigate = jest.fn();
jest.mock('ui/hooks', () => ({
  useNavigate: () => {
    return {
      navigate: mockNavigate,
    };
  },
}));

jest.mock('ui/router/RouteContext', () => {
  return {
    useRouter: () => {
      return {
        params: { web3_wallet_id: 'test-id' },
        resolve: () => {
          return {
            toURL: {
              href: 'https://www.ssddd.com',
            },
          };
        },
      };
    },
  };
});

jest.mock('ui/contexts', () => {
  return {
    useCoreUser: (): Partial<UserResource> => {
      return {
        primaryWeb3WalletId: 'test-id',
        web3Wallets: [
          {
            id: 'test-id',
            web3Wallet: '0x0000000000000000000000000000000000000000',
            verification: { status: 'verified' },
          } as Web3WalletResource,
        ],
      };
    },
    useEnvironment: jest.fn(
      () =>
        ({
          userSettings: {} as PartialDeep<UserSettingsResource>,
        } as PartialDeep<EnvironmentResource>),
    ),
  };
});

describe('<Web3WalletDetail/>', () => {
  it('renders Web3WalletDetail', () => {
    const tree = renderJSON(<Web3WalletDetail />);
    expect(tree).toMatchSnapshot();
  });
});
