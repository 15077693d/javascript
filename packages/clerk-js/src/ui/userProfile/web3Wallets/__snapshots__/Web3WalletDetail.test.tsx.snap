// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Web3WalletDetail/> renders Web3WalletDetail 1`] = `
Array [
  <div
    className="cl-page-heading"
  >
    <a
      aria-label="Back Button"
      className="cl-back-button cl-back-button"
      onClick={[Function]}
    >
      <svg
        data-filename="../shared/assets/icons/arrow-left.svg"
      />
    </a>
    <div
      className="cl-text-container"
    >
      <h2
        className="heading h2 cl-title"
      >
        Web3 wallet
      </h2>
      <p
        className="cl-subtitle"
      >
        Manage this Web3 wallet
      </p>
    </div>
  </div>,
  <div
    className="card card cl-themed-card cl-list-card cl-verifiable-field-card"
  >
    <div>
      <div
        className="cl-list-item elementContainer"
        disabled={false}
      >
        <div
          className="listItem line"
        >
          <div
            className="start"
          >
            <div>
              <div
                className="cl-ident-detail"
              >
                0x0000000000000000000000000000000000000000
                <div
                  className="cl-tags"
                >
                  <span
                    className="tag primary sm cl-tag"
                  >
                    Primary
                  </span>
                  <span
                    className="tag success sm cl-tag"
                  >
                    Verified
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="iconContainer"
          />
        </div>
      </div>
      <div
        className="cl-list-item elementContainer"
        disabled={false}
      >
        <div
          className="listItem line"
        >
          <div
            className="start"
          >
            <div
              className="cl-primary-status-container"
            >
              <div
                className="cl-description"
              >
                <svg
                  data-filename="../shared/assets/icons/mail.svg"
                />
                <div
                  className="cl-text"
                >
                  <div
                    className="cl-title"
                  >
                    Primary web3 wallet
                  </div>
                </div>
              </div>
              <label
                className="container disabled"
                onKeyPress={[Function]}
              >
                <span
                  aria-checked={true}
                  className="toggle"
                  role="switch"
                  tabIndex={0}
                >
                  <input
                    checked={true}
                    disabled={true}
                    name="primary_toggle"
                    onChange={[Function]}
                    tabIndex={-1}
                    type="checkbox"
                  />
                  <span
                    className="slider"
                  />
                </span>
              </label>
            </div>
          </div>
          <div
            className="iconContainer"
          />
        </div>
      </div>
    </div>
    <button
      className="button text error hoverable cl-add-resource-button cl-remove-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        data-filename="../shared/assets/icons/bin.svg"
      />
      Unlink
    </button>
  </div>,
]
`;
